django/db/backends/__init__.py

```diff
@@ -823,6 +823,15 @@ def cache_key_culling_sql(self):
         """
         return "SELECT cache_key FROM %s ORDER BY cache_key LIMIT 1 OFFSET %%s"
 
+    def case_cast_sql(self, db_type, internal_type):
```

Comment:
Is the cast SQL returned by this method actually specific to `CASE` statements in some way? Are values in `CASE` statements cast differently from other casts in SQL?